OpenSim/Common/Component.h

```diff
@@ -721,6 +721,10 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
      */
     template <class C = Component>
     const C& getComponent(const std::string& pathname) const {
+        return getComponent<C>(ComponentPath(pathname));
+    }
+    template <class C = Component>
```

Comment:
Is there a chance the constructor/conversion from pathname to ComponentPath fails? If so what's returned? Is there an InvalidComponentPath? If this is wired to user input at some point we need to account for these variations, maybe done already elsewhere.

Reply:
> Is there a chance the constructor/conversion from pathname to ComponentPath fails?

@aymanhab yes, an `OpenSim::Exception` can be thrown but the current code base still creates a `ComponentPath` from the provided `std::string`; it's just delayed until `traversePathToComponent()`.

> If so what's returned?

The function doesn't return.

> Is there an InvalidComponentPath?

Nope; but something like `ComponentNotFoundOnSpecifiedPath` might get thrown.

> If this is wired to user input at some point we need to account for these variations, maybe done already elsewhere.

By "wired to user input," do you mean in the GUI? I'm not sure what you mean.

Reply:
@aseth1 I mean if this is expected to be called directly by users through C++ API or Specified in XML for example, then this code should be ready to handle misformed "ComponentPath" String.

Reply:
An exception will be thrown if the pathname has invalid characters or is ill-formed, or if a component does not exist at the specified path. Is there another way that invalid user input should be handled?

Reply:
Thanks @chrisdembia an exception is perfectly reasonable, and I'm assuming whoever calls this somehow knows that it may throw and can handle it. Thanks for the clearup.